### ISSUE TITLE
Video playback

### ISSUE THREAD

**Issue by wheybags** [wheybags/freeablo#27]
ffmpeg is probably what we want here, but I'm no expert

**Comment by konopka90** [comment]
http://dranger.com/ffmpeg/

Tutorial for SDL 1.2

**Comment by wheybags** [comment]
I think we might actually consider using https://wiki.videolan.org/LibVLC/ instead.
As far as I understand, it's a lot higher level, and easier to use.
Plus, ffmpeg is an absolute pain to use, with it's two almost-compatible-but-not-quite forks

**Comment by konopka90** [comment]
I'm not sure but LibVLC can't open SMK videos. Latest VLC can only play sound but no video :(

**Comment by konopka90** [comment]
Can be useful https://github.com/pockethook/player

**Comment by wheybags** [comment]
Hmm, yeah, vlc is just giving me audio too :s
That player crashes for me when I try to open an smk file though :s
```
[Arts-MacBook-Pro] ~/player [master] >
$ ./player ~/freeablo/build/logo.smk
[swscaler @ 0x7fb07a829600] bad dst image pointers
[swscaler @ 0x7fb07a829600] bad dst image pointers
[swscaler @ 0x7fb07a829600] bad dst image pointers
[swscaler @ 0x7fb07a829600] bad dst image pointers
[swscaler @ 0x7fb07a829600] bad dst image pointers
[swscaler @ 0x7fb07a829600] bad dst image pointers
Dis[swscaler @ 0x7fb07a829600] play error: SDL textbad dst image pointers
ure update
```

ffmpeg can definitely be made to work anyway, someone did a half baked implementation that got abandoned a while ago (actually, was that you? :p)

**Comment by konopka90** [comment]
yup that was me ;p there was audio and video but without synchronization what is core of the problem. I'll try to debug this player, maybe it will work with some changes.

**Comment by wheybags** [comment]
It worked fine on an mp4 file btw

**Comment by konopka90** [comment]
Sounds good

**Comment by konopka90** [comment]
Fail - it is possible to play SMK with small changes but this player doesn't support sound :-1:

**Comment by Hydroque** [comment]
https://trac.videolan.org/vlc/ticket/17320
Check out this! It seem like master of vlc fully supports SMK files!

**Comment by mewmew** [comment]
>  Check out this! It seem like master of VLC fully supports SMK files!

Wow, that's great! Never thought such an ancient format would be supported upstream :p

**Comment by wheybags** [comment]
Nice! Iirc libvlc is much easier to use than ffmpeg too

**Comment by ChefKeeper** [comment]
Tryed yesterday with latest vlc version : 2-2-8, the smk file reads and play with audio and video but there's still a problem with the colors and terminal gets spammed with errors about alpha.

**Comment by Hydroque** [comment]
What is the problem with the alpha channel? We can do a patch if we need it. Would be muuuch easier.
